frontend/apps/reader/readerui.lua

```diff
@@ -597,6 +597,24 @@ function ReaderUI:doShowReader(file, provider)
         ReaderUI.instance:onClose()
     end
     local document = DocumentRegistry:openDocument(file, provider)
+
+    -- wrap getCurrentPos method with setIgnoreInput's, to avoid ANRs
+    if Device:isAndroid() then
+        local wrapped_getCurrentPos = document.getCurrentPos
+        document.getCurrentPos = function(...)
```

Comment:
I'm really not fond of this, why do you feel the need for this vs. what you had previously ?

Reply:
Please see my post below.